lib/engines/content_object_store/features/schedule_object.feature

```diff
@@ -0,0 +1,25 @@
+Feature: Schedule a content object
+  Background:
+    Given the content object store feature flag is enabled
+    And I am a GDS admin
+    And the organisation "Ministry of Example" exists
+    And a schema "email_address" exists with the following fields:
+      | field         | type   | format | required |
+      | email_address | string | email  | true     |
+    And an email address content block has been created
+
+
+  Scenario: GDS Editor schedules a content object
+    When I am updating a content block
+    Then I am asked when I want to publish the change
+    When I choose to schedule the change
+    And I enter a date 7 days in the future
+    And I accept and publish
+    Then the edition should have been scheduled successfully
+    And I should be taken back to the document page
+    And I should see the scheduled date on the object
+
+#  Scenario: A scheduled content object is published
```

Comment:
would also like to investigate whether it's possible to do this test - setup a scheduling job in the worker, fast forward the time and see that it gets published. But not sure if possible....

Reply:
IIRC we can include [sidekiq/testing](https://github.com/sidekiq/sidekiq/wiki/Testing#testing-delay-and-delay_for-methods) which enables:

1. We can check that our code correctly puts the right job on the right delayed queue with the right time
2. We can then run with inline mode which will process all jobs straight away regardless of what our code says

Reply:
Nice, I'm using the inline method in a feature test to check it gets published, and have tested that we are pushing the right data to the right queue in the integration test `worker_test.rb`